apps/src/code-studio/components/progress/teacherPanel/StudentTable.jsx

```diff
@@ -53,7 +53,18 @@ class StudentTable extends React.Component {
 
     // Sort students, in-place.
     isSortedByFamilyName
-      ? students.sort((a, b) => a.familyName.localeCompare(b.familyName))
+      ? students.sort((a, b) => {
+          if (a.familyName && b.familyName) {
+            return a.familyName.localeCompare(b.familyName);
+          } else if (a.familyName) {
+            // Sort empty family names last.
+            return -1;
+          } else if (b.familyName) {
+            return 1;
+          } else {
+            return 0;
+          }
+        })
```

Comment:
The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) for `String.prototype.localeCompare()` state:

> When comparing large numbers of strings, such as in sorting large arrays, it is better to create an [Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator) object and use the function provided by its [compare()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/compare) method.

Indeed, when using this collator we seem to get null checking and diacritical mark sorting for free.

![image](https://github.com/code-dot-org/code-dot-org/assets/4108328/84da2ce1-df06-4c62-a16f-4dc0b712c47f)


```suggestion
      // Can specify a locale here if known, otherwise uses system default.
      const collator = new Intl.Collator();
      isSortedByFamilyName
        ? students.sort((a, b) => collator.compare(a?.familyName, b?.familyName)
        : students.sort((a, b) => collator.compare(a?.name, b?.name))
```

Reply:
nice! this is much more readable